iceoryx_posh/source/runtime/ipc_interface_base.cpp

```diff
@@ -59,9 +59,10 @@ std::string IpcMessageErrorTypeToString(const IpcMessageErrorType msg) noexcept
     return cxx::convert::toString(static_cast<std::underlying_type<IpcMessageErrorType>::type>(msg));
 }
 
-IpcInterfaceBase::IpcInterfaceBase(const RuntimeName_t& runtimeName,
-                                   const uint64_t maxMessages,
-                                   const uint64_t messageSize) noexcept
+template <typename IpcChannelType>
```

Comment:
Actually this would make the code more generic then I want it to be. I can implement template code in a cpp file when I explicitly instantiate it afterwards and then the template provides only an implementation for the types I explicitly instantiate it with.
This is exactly what I want here.

See the lines 214 - 216 where I instantiate it!

The code belongs only in the `.inl` file when I as an implementer do not know all the types the template can be instantiate with. Here I just want to support 3 types I know at compile time and do not want to extend the support to other types.